rs/p2p/quic_transport/src/connection_handle.rs

```diff
@@ -42,39 +42,34 @@ impl Drop for SendStreamDropGuard {
 #[derive(Clone, Debug)]
 pub struct ConnectionHandle {
     pub connection: Connection,
-    metrics: QuicTransportMetrics,
-    conn_id: ConnId,
+    pub metrics: QuicTransportMetrics,
+    pub conn_id: ConnId,
 }
 
 impl ConnectionHandle {
-    pub fn new(connection: Connection, metrics: QuicTransportMetrics, conn_id: ConnId) -> Self {
-        Self {
-            connection,
-            metrics,
-            conn_id,
-        }
-    }
-
-    pub fn conn_id(&self) -> ConnId {
-        self.conn_id
-    }
-
-    /// Performs an RPC operation on the already established connection.
+    /// Executes an RPC operation over an already-established connection.
+    ///
+    /// This method leverages the QUIC transport layer, which continuously monitors the connection’s health
+    /// and automatically attempts reconnection as necessary. As a result, any errors returned by this method
+    /// should be considered transient (retryable).
+    ///
+    /// In this P2P architecture, where there is a designated dialer and receiver, connection management
+    /// is delegated solely to the transport layer. This differs from typical client-server architectures,
+    /// where connections can be managed directly by the caller.
     ///
-    /// Since the QUIC transport layer continuously monitors connection health and automatically reconnects as needed,
-    /// all errors returned by this method can be safely treated as transient (retryable).
-    /// Therefore, we return an anyhow::Error for error handling.
+    /// Note: This method is cancel-safe.
```

Comment:
Did you talk to the Quinn maintainers about the methods we call int his method?

Reply:
Yes, they said it should be cancel safe. But I guess no guarantees.
If they really did it in a sans-io way as it is [claimed in this blog](https://www.firezone.dev/blog/sans-io) then ideally things will work.

I am still undecided if I will do the channel approach we discussed. I will clean up the code a bit and then decide if there is high risk if we can end up in corrupted state if we abort.